sdk/src/Core/Amazon.Runtime/Pipeline/HttpHandler/_mobile/HttpRequestMessageFactory.cs

```diff
@@ -268,29 +268,36 @@ private static HttpClient CreateManagedHttpClient(IClientConfig clientConfig)
 #if NETSTANDARD
             if (clientConfig.MaxConnectionsPerServer.HasValue)
                 httpMessageHandler.MaxConnectionsPerServer = clientConfig.MaxConnectionsPerServer.Value;
-#endif
-
-            // If HttpClientHandler.AllowAutoRedirect is set to true (default value),
-            // redirects for GET requests are automatically followed and redirects for POST
-            // requests are thrown back as exceptions.
-            // If HttpClientHandler.AllowAutoRedirect is set to false (e.g. S3),
-            // redirects are returned as responses.
-            httpMessageHandler.AllowAutoRedirect = clientConfig.AllowAutoRedirect;
 
-            // Disable automatic decompression when Content-Encoding header is present
-            httpMessageHandler.AutomaticDecompression = DecompressionMethods.None;
-
-            var proxy = clientConfig.GetWebProxy();
-            if (proxy != null)
+            try
             {
-                httpMessageHandler.Proxy = proxy;
-            }
+#endif
+                // If HttpClientHandler.AllowAutoRedirect is set to true (default value),
+                // redirects for GET requests are automatically followed and redirects for POST
+                // requests are thrown back as exceptions.
+                // If HttpClientHandler.AllowAutoRedirect is set to false (e.g. S3),
+                // redirects are returned as responses.
+                httpMessageHandler.AllowAutoRedirect = clientConfig.AllowAutoRedirect;
+                // Disable automatic decompression when Content-Encoding header is present
+                httpMessageHandler.AutomaticDecompression = DecompressionMethods.None;
+
+                var proxy = clientConfig.GetWebProxy();
```

Comment:
I'm thinking we should wrap the proxy settings in a separate try/catch block in case the SDK is running on a platform where AllowAutoRedirect & AutomaticDecompression are not support but the proxy settings are.